t/test_rsync_loc_handlers.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+# Tests behaviour of rose/loc_handlers/rsync_remote_check.py against many
+# versions of Python.
+
+set -eu
+
+python --version
+
+mkdir -p foo/baz
+echo "HI" > foo/bar
+echo "THERE" > foo/baz/qux
+
+python metomi/rose/loc_handlers/rsync_remote_check.py $PWD/foo 'blob' 'tree' > out.txt
+test="it prints tree"
+grep tree out.txt > /dev/null && echo "PASS ${test}" || exit 1
+
+test="it prints foo/baz/qux"
+grep "foo/baz/qux" out.txt > /dev/null && echo "PASS ${test}" || exit 1
+
+test="it prints 4 lines"
+test "$(cat out.txt | wc -l)" == 4 && echo "PASS ${test}" || exit 1
```

Comment:
How about performing a known-good-output test here rather than grepping for individual words. I think the output of this command should be stable (possibly access mode aside although that can be controlled when the files are created at the start of the test).